plugins/catalog-backend/src/database/CommonDatabase.ts

```diff
@@ -519,6 +548,46 @@ export class CommonDatabase implements Database {
   }
 }
 
+function parsePagination(
+  input?: EntityPagination,
+): { limit?: number; offset?: number } {
+  if (!input) {
+    return {};
+  }
+
+  let { limit, offset } = input;
+
+  if (input.after !== undefined) {
+    let cursor;
+    try {
+      const json = Buffer.from(input.after, 'base64').toString('utf8');
+      cursor = JSON.parse(json);
+    } catch {
+      throw new InputError('Malformed after cursor');
```

Comment:
Prolly give each of these some kind of uniqueness so we can figure out what's up if someone's hitting this error

Reply:
Ah yeah, added